app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/NationalInsuranceController.scala

```diff
@@ -62,8 +82,23 @@ class NationalInsuranceController @Inject()(
           .fold(
             formWithErrors => Ok(html.national_insurance_number(formWithErrors)),
             validNino =>
-              updateSessionAndRedirect(existingSession.copy(nino = Some(validNino)))(
-                routes.DateOfBirthController.showDateOfBirthForm())
+              if (agent.authNino.flatMap(normalizeNino) == normalizeNino(validNino.value)) {
```

Comment:
Please move this predicate to a single line method with a name which reflects what the comparison means